src/serializer/functions.js

```diff
@@ -122,40 +122,69 @@ export class Functions {
     };
   }
 
-  checkReactRootComponents(statistics: ReactStatistics, react: ReactSerializerState): void {
+  _generateWriteEffectsForReactComponentTree(
+    componentType: ECMAScriptSourceFunctionValue,
+    effects: Effects,
+    simpleClassComponents: Set<Value>
+  ): void {
+    if (this.realm.react.output === "bytecode") {
+      throw new FatalError("TODO: implement React bytecode output format");
+    } else {
+      let additionalFunctionEffects = this._createAdditionalEffects(effects);
+      let value = effects[0];
+
+      if (value === this.realm.intrinsics.undefined) {
```

Comment:
Oh, I see, it gets used here. It is a bit confusing to me that we use `undefined` for this (even though technically we ignore its "meaning" and just use it as a marker). Could this be something more special?

Reply:
I didn't want to do anything special, because in the past this has caused more confusion than necessary.

Reply:
Sounds good, thanks for explaining